src/services/products.js

```diff
@@ -0,0 +1,110 @@
+const { products } = require('../db');
+
+async function createProduct(req) {
+  const { type, color, price = 0, quantity = 1 } = req.body;
+
+  if (!type) {
+    const err = new Error('ERROR: No product type defined');
+    err.status = 400;
+    throw err;
+  }
+  if (!color) {
+    const err = new Error('ERROR: No product color defined');
+    err.status = 400;
+    throw err;
+  }
+
+  const product = {
+    type,
+    color,
+    price: parseInt(price, 10),
+    quantity: parseInt(quantity, 10),
```

Comment:
When price or quantity are invalid the NaN value will be assigned. Should throw an error instead.